### PR TITLE
Add PreferUnquoted emitter string manipulator

### PR DESCRIPTION
**Pull request by duncanthomson**
The intention of this PR is to provide a simple way to decide, based on the content of a string, how to present that string in YAML. In particular, we're trying to avoid having to quote a string or escape characters within it.

The yaml-cpp library doesn't presently provide a way to do this. There are a couple of ways you could achieve this "properly":

1. Expose functions like `IsValidPlainScalar` publicly, so a client can make a determination of which string manipulator they want to use based on the results of those calls, or

2. Provide a mechanism to express ordered preference of string manipulators, e.g. Ideally, I'd like to leave a string unquoted, but if that's not possible (given the content of the string), I'd like to present the string as a Literal, but if that's not possible, I'd like to fallback to using double quotes and escape the content of the string as required.

This PR adds a new manipulator `PreferUnquoted` which applies the ordered preferences given in the example of (2) above, but in a hard-coded fashion. This is a quick and easy way to satisfy our immediate requirements, but isn't suitable for pushing upstream. For that, we'd need to re-work this to be user-configurable.

### CODE REVIEW

File: src/emitter.cpp [hunk 1]
```diff
@@ -702,6 +702,9 @@ Emitter& Emitter::Write(const std::string& str) {
       Utils::WriteLiteralString(m_stream, str,
                                 m_pState->CurIndent() + m_pState->GetIndent());
       break;
+    case StringFormat::PreferUnquoted:
+      // error
```
Comment (chri7325) [inline src/emitter.cpp]: If this is an error, should something happen to alert us?

**Review by chri7325 — APPROVED** [review]
Minor question.  Rest looks good to me.

File: src/emitter.cpp (hunk 1 above)
Comment (duncanthomson) [reply src/emitter.cpp]: That's a fair point. I've replaced the comment with an `assert()` to demonstrate that we shouldn't get here (and alert us if we do).

Comment (duncanthomson) [reply src/emitter.cpp]: Duh! Your observation points out that PreferUnquoted doesn't need to be in the StringFormat enum at all. Thank you @chri7325

**Review by chri7325 — APPROVED** [review]

**Pull request merged**